File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentPiercing.java
package thedarkcolour.futuremc.enchantment;

import net.minecraft.enchantment.Enchantment;

public class EnchantmentPiercing extends EnchantmentCrossbowBase {
    public EnchantmentPiercing() {
        super(Rarity.COMMON, "piercing");
    }

    @Override
    public int getMaxLevel() {
        return 4;
    }

    @Override
    public int getMinEnchantability(int enchantmentLevel) {
        return 1 + (enchantmentLevel - 1) * 10;
    }

    @Override
    protected boolean canApplyTogether(Enchantment ench) {
        return super.canApplyTogether(ench) && ench != Enchantments.MULTISHOT;
    }
}

File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantmentPiercing.kt
package thedarkcolour.futuremc.enchantment

import net.minecraft.enchantment.Enchantment
import thedarkcolour.futuremc.registry.FEnchantments

class EnchantmentPiercing : EnchantmentCrossbowBase(Rarity.COMMON, "piercing") {
    override fun getMaxLevel(): Int {
        return 4
    }

    override fun getMinEnchantability(enchantmentLevel: Int): Int {
        return 1 + (enchantmentLevel - 1) * 10
    }

    override fun canApplyTogether(ench: Enchantment): Boolean {
        return super.canApplyTogether(ench) && ench !== FEnchantments.MULTISHOT
    }
}